sysdeps/managarm/generic/file.cpp

```diff
@@ -1380,18 +1380,22 @@ int sys_poll(struct pollfd *fds, nfds_t count, int timeout, int *num_events) {
 
 	managarm::posix::SvrResponse<MemoryAllocator> resp(getSysdepsAllocator());
 	resp.ParseFromArray(recv_resp->data, recv_resp->length);
-	__ensure(resp.error() == managarm::posix::Errors::SUCCESS);
-	__ensure(resp.events_size() == count);
+	if(resp.error() == managarm::posix::Errors::ILLEGAL_ARGUMENTS) {
+		return EINVAL;
+	} else {
```

Comment:
this else isn't necessary, it's only effect is adding an unneeded indentation level (early-return already has the effect of skipping the subsequent code)

Reply:
While true, I like the clearness it gives. It's also not breaking the current code style as a lot of the sysdeps do this too. Personally, I'd keep it as is unless asked to do otherwise.

Reply:
we have always preferred early return, though? also, this adds no clarity, only a bigger patch

Reply:
odd thing to nitpick